### ISSUE TITLE
Feature request for diskdrive read test (or other suitable location)

### ISSUE THREAD

**Issue by peoyli** [keirf/amiga-stuff#70]
**Add start/stop and seek (preferrably directly to a position).**

When I use the floppy read test, I have a (tiny) bunch of floppys with no or different grade of errors on them to test with. Usually they show the same (amount and location) of unreadable sectors, sometime the result surprises me by giving a fully readable marked-bad floppy :)

Between the tests and after the bunch has gone through, I use a cleaning disk with IPA, so preferrably a different track on the cleaning disk should be used every time to eveb the wear on the cleaning disk and not cleaning with a dirty location.

**Idea how it could be implemented (but of course, you decide on that)**
When starting a read test without a floppy inserted, do not start trying to read, just make it ready for manual start (this could be called "cleaning mode" or described as "read-test on single track"). Automatic seek should be disabled.

Seek: click on any of the tracks to directly seek to that location (might even be useful in normal read mode with automatic stepping to next track)
When (cleaning) floppy has been inserted, start/stop to start the spindle (cleaning/reading), seek as previous (could be used to do the most of the cleaning on one track and wipe dry on another).

**Comment by keirf** [comment]
Greaseweazle has a cleaning mode (inspired by other tools' cleaning modes) which actively seeks around during cleaning in a "scrubbing" kind of fashion.

I could implement this very easily in ATK. A new cleaning test screen, with start/stop function key. Default stopped. 

WDYT?

**Comment by keirf** [comment]
Oh, it could also automatically stop after say 3 passes over the cleaning disk, or something like that.

**Comment by peoyli** [comment]
Would be very useful to have the cleaning function on ATK. So maybe use the same as in Greaseweazle and add manual seek later ? The cleaning floppy I have have 80 boxes on the label, made for marking which tracks have been used (it was probably meant for "single use 80 times"). Still useful to clean on multiple tracks in one session, since the head distance/pressure on the disk could be different at different positions.

**Comment by keirf** [comment]
Yeah I think with automated cleaning it would make good use of the cleaning disk without wearing a particular track.

**Comment by screwtop** [comment]
I like this idea too, and gave found the Greaseweazle cleaning mode handy. An interim method with ATK is just to use the drive signal test and control the motor and stepping manually. :)
